src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCache.cs

```diff
@@ -211,9 +211,68 @@ public bool TryGetValue(object key, out object? result)
             DateTime utcNow = UtcNow;
```

Comment:
Does this need to be done before the TryGetValue call, or could it be consolidated into your PostProcessTryGetValue helper?

Reply:
It is a valid question; my primary aim here was to retain the existing behaviour, but I think it is fair to challenge it - I suspect the only *critical* thing is that `utcNow` doesn't change between the 4 places where it is used (and as a secondary aim: that we don't pay the cost of fetching it 4 times), but as long as we *do* check expiration, I can't think of any way in which it is less "valid" if we defer the time lookup to after the initial fetch. I'll try moving it.